docs/core/testing/unit-testing-mstest-writing-tests-testcontext.md

```diff
@@ -0,0 +1,183 @@
+- As a property of a test class. In this context, the properties related to the test run are available.
+- As a constructor parameter of a test class (starting with v3.6) as an alternative to the property. This construct is recommended over the property because it gives access to the object in the constructor (while the property is only available after the constructor has run). It's also helping to ensure immutability of the object and finally it helps the compiler to enforce that the object is not null.
+
+```csharp
+[TestClass]
+public class MyTestClass
+{
+    public TestContext TestContext { get; set; }
+
+    [AssemblyInitialize]
+    public static void AssemblyInitialize(TestContext context)
+    {
+        // Access TestContext properties and methods here. The properties related to the test run are not available.
+    }
+
+    [ClassInitialize]
+    public static void ClassInitialize(TestContext context)
+    {
+        // Access TestContext properties and methods here. The properties related to the test run are not available.
+    }
+
+    [TestMethod]
+    public void MyTestMethod()
+    {
+        // Access TestContext properties and methods here
+    }
+}
+```
+
+or with MSTest 3.6+
+
+```csharp
+[TestClass]
+public class MyTestClass
+{
+    private readonly TestContext _testContext;
+
+    public MyTestClass(TestContext testContext)
+    {
+        _testContext = testContext;
+    }
+
+    [AssemblyInitialize]
+    public static void AssemblyInitialize(TestContext context)
+    {
+        // Access TestContext properties and methods here. The properties related to the test run are not available.
+    }
+
+    [ClassInitialize]
+    public static void ClassInitialize(TestContext context)
+    {
+        // Access TestContext properties and methods here. The properties related to the test run are not available.
+    }
+
+    [TestMethod]
+    public void MyTestMethod()
+    {
+        // Access TestContext properties and methods here
+    }
+}
+```
+
+## The `TestContext` members
+
+The `TestContext` class provides properties about the test run along with methods to manipulate the test environment. In this section, we will cover the most commonly used properties and methods.
+
+### Test run information
+
+The `TestContext` provides information about the test run, such as:
+
+- `TestName` – the name of the currently executing test.
+- `CurrentTestOutcome` – the result status of the current test.
+- `FullyQualifiedTestClassName` – the full name of the test class.
+- `TestRunDirectory` – the directory where the test run is executed.
+- `DeploymentDirectory` – the directory where the deployment items are located.
+- `ResultsDirectory` – the directory where the test results are stored.  Typically a subdirectory of the `TestRunDirectory`.
+- `TestRunResultsDirectory` – the directory where the test results are stored. Typically a subdirectory of the `ResultsDirectory`.
```

Comment:
How is it different to `ResultsDirectory` and `TestResultsDirectory`?

Reply:
I have looked at the code and it's not clear how they differ but we do have the 2 properties.